src/main/java/com/google/sps/Scheduler.java

```diff
@@ -1,15 +1,113 @@
 package com.google.sps;
 
-import java.util.Collection;
-
 import com.google.sps.data.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
 
 /**
- * Static class for permuting schedules given a list of courses.
+ * Static class for permuting schedules given a list of courses and a set of Invariants.
  */
 public final class Scheduler {
-    public Collection<Schedule> generateSchedules(Collection<Course> courses, Invariants invariants){
-        //TODO: This
-        return null;
+  /**
+   * Generates all valid schedules given a list of courses and invariant
+   * constraints.
+   * 
+   * @param courses    The list of courses to generate the schedules from.
+   * @param invariants The invariants that constrain the valid schedules
+   * @return A list of valid schedules.
+   */
+  public List<Schedule> generateSchedules(List<Course> courses, Invariants invariants) {
+    ArrayList<Course> betterCourses = new ArrayList<>(courses);
+    Collections.sort(betterCourses, Course.REQUIRED_TO_FRONT);
+    Set<List<Course>> schedulesSet = new HashSet<List<Course>>();
```

Comment:
extreme nit: `schedulesSet` -> `scheduleSet`

I don't think most people care about something like this if I'm honest, so feel free to ignore if you prefer it as is. When I've got a collection of foobars, I'll either use `foobars` or `foobarList`, `foobarSet`, etc. for variable names.